capstone/src/main/java/com/google/step/servlets/PageViewServlet.java

```diff
@@ -0,0 +1,132 @@
+     *                int year, String restaurantKey -> getYearRestaurantPageViews()
+     *                empty -> getAllPageViews();
+     *                A SC_FORBIDDEN status will be set if none of these three specified combinations
+     *                is present.
+     * @param response Json String response that varies depending of the method called by the mix of
+     *                 parameters in the request.
+     *                 getCurrentPageViews() -> WeeklyPageView (as Json)
+     *                 getYearRestaurantPageViews() -> List<WeeklyPageView> (as Json list)
+     *                 getAllPageViews() -> List<RestaurantPageViews> (as Json list)
+     * @throws IOException can be thrown by the writer for the response.
+     */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String restaurantKey;
+        if (request.getParameter("restaurantKey") != null) {
+            restaurantKey = request.getParameter("restaurantKey");
+        } else {
+            restaurantKey = null;
+        }
+
+        int year = 0;
+        if (request.getParameter("year") != null) {
+            year = Integer.parseInt(request.getParameter("year"));
+        }
+
+        String json;
+        if (restaurantKey == null && year == 0) {
+            // getAllPageViews()
+            List<RestaurantPageViews> restaurantPageViewsList = metricsClient.getAllPageViews();
+            json = convertRestaurantPageViewListToJsonUsingGson(restaurantPageViewsList);
+        } else if (year == 0 && restaurantKey != null) {
+            // getCurrentPageViews()
+            WeeklyPageView weeklyPageView = metricsClient.getCurrentPageViews(restaurantKey);
+            json = convertWeeklyPageViewToJsonUsingGson(weeklyPageView);
+
+        } else if (year != 0 && restaurantKey != null) {
+            // getYearRestaurantPageViews
+            List<WeeklyPageView> weeklyPageViewList = metricsClient.getYearRestaurantPageViews(year, restaurantKey);
+            json = convertWeeklyPageViewListToJsonUsingGson(weeklyPageViewList);
+        } else {
+            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+            return;
+        }
+
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
+    }
+
+    /**
+     *
+     * @param request should contain the <String, restaurantKey> and <String, restaurantName> as
+     *                parameters. A SC_FORBIDDEN status can be sent to the frontend if one of these
+     *                parameters is not present.
+     * @param response no response.
+     * @throws IOException can be thrown by the response.
+     */
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String restaurantKey = request.getParameter("restaurantKey");
+        String restaurantName = request.getParameter("restaurantName");
```

Comment:
Update this as the name parameter is no longer passed as a param from the frontend.

Reply:
Yup I'm just waiting for the other PR to be accepted and rebase the changes.

Reply:
Alright done!